packages/editor/src/components/rich-text/index.js

```diff
@@ -936,23 +994,28 @@ const RichTextContainer = compose( [
 		if ( ownProps.isSelected === true ) {
 			return {
 				isSelected: context.isSelected,
+				clientId: context.clientId,
 			};
 		}
 
 		// Ensures that only one RichText component can be focused.
 		return {
 			isSelected: context.isSelected && context.focusedElement === ownProps.instanceId,
 			setFocusedElement: context.setFocusedElement,
+			clientId: context.clientId,
 		};
 	} ),
-	withSelect( ( select ) => {
+	withSelect( ( select, props ) => {
 		const { isViewportMatch } = select( 'core/viewport' );
-		const { canUserUseUnfilteredHTML, isCaretWithinFormattedText } = select( 'core/editor' );
+		const { canUserUseUnfilteredHTML, isCaretWithinFormattedText, getAnnotationsForRichText } = select( 'core/editor' );
+
+		const annotations = getAnnotationsForRichText( props.clientId );
```

Comment:
This retrieves the same annotations for all the RichText instances of any block which is probably wrong. Related https://github.com/WordPress/gutenberg/pull/7718/files#r230091967

Reply:
Yeah, you can have multiple instances of `RichText` in one block. Check Gallery, Quote or Pullquote blocks as examples.